stable/appmesh-gateway/templates/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "appmesh-gateway.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "appmesh-gateway.labels" . | indent 4 }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "appmesh-gateway.fullname" . }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "appmesh-gateway.fullname" . }}
+        app.kubernetes.io/component: "appmesh-gateway"
+      annotations:
+        {{- if .Values.appmesh.preview }}
+        appmesh.k8s.aws/preview: "enabled"
+        {{- end }}
+        {{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "appmesh-gateway.serviceAccountName" . }}
+      terminationGracePeriodSeconds: 30
+      {{- if .Values.podAntiAffinity }}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app.kubernetes.io/name: {{ include "appmesh-gateway.name" . }}
+                topologyKey: kubernetes.io/hostname
+              weight: 100
+      {{- end }}
+      containers:
+      - name: envoy
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        ports:
+        - containerPort: 8088
+          name: http
+          protocol: TCP
+        - containerPort: 9901
+          name: http-prom
+          protocol: TCP
+        livenessProbe:
+          tcpSocket:
+            port: http-prom
+        readinessProbe:
+          initialDelaySeconds: 5
+          tcpSocket:
+            port: http-prom
```

Comment:
Both of these will be considered "good" pretty much as soon as the Envoy starts, given it's the admin port. We'd probably want to clarify this with an actual check for bootstrapped configuration, e.g. [this command](https://github.com/aws/aws-app-mesh-examples/blob/master/walkthroughs/tls-with-acm/infrastructure/ecs-service.yaml#L116-L118).

Reply:
Wouldn't that cause all pods to crash loop when the App Mesh xDS server has an outage? My assumption is that Envoy should keep working with the last know configuration, do we want to have a cascading failure?

Reply:
So long as the Envoy has bootstrapped the initial configuration, it will respond back with the `LIVE` state. This is true even if the xDS server has crashed and Envoy is retrying/backing off on the connection. 

It would be the case that an Envoy which cannot bootstrap due to an xDS outage would be considered unhealthy, but that would be expected.

Reply:
Ok I've added the live check to the liveness probe, thanks for the explanation.